lib/src/native/realm_core.dart

```diff
@@ -3399,25 +3383,30 @@ extension PlatformEx on Platform {
 /// @nodoc
 class SyncErrorDetails {
   final String message;
-  final SyncErrorCategory category;
+  final int _categoryFlag;
   final int code;
-  final String? detailedMessage;
+  final String? path;
   final bool isFatal;
   final bool isClientResetRequested;
   final String? originalFilePath;
   final String? backupFilePath;
   final List<CompensatingWriteInfo>? compensatingWrites;
-  const SyncErrorDetails(
+  SyncErrorDetails(
     this.message,
-    this.category,
```

Comment:
This is removed, but errors still have categories. And one error can have multiple categories. Kotlin is including this in their errors as detail message property.

Reply:
`category` argument is removed.